### PR TITLE
Make training job 'pause' if there aren't enough new games.

### PR DESCRIPTION
**Pull request by amj**
Adds function to rl_loop.py to check number of games finished since last training was complete.  Check that number before beginning the next training job.  It should ideally be ~100% the size of the generation, but if not, sleep a bit before training again.

### CODE REVIEW

File: rl_loop.py [hunk 1]
```diff
@@ -33,6 +34,8 @@
 SGF_DIR = os.path.join(BASE_DIR, 'sgf')
 TRAINING_CHUNK_DIR = os.path.join(BASE_DIR, 'data', 'training_chunks')
 
+GAMES_BEFORE_TRAINING = 4000
```
Comment (Kashomon) [inline rl_loop.py]: I'm big on comments, of course. Can we add one here?

File: rl_loop.py [hunk 2]
```diff
@@ -49,20 +52,36 @@ def print_flags():
                     for flag, value in flags.items()))
 
 
+def get_models():
```
Comment (Kashomon) [inline rl_loop.py]: `+` method doc

File: rl_loop.py [hunk 3]
```diff
@@ -49,20 +52,36 @@ def print_flags():
...
-    model_filenames = [os.path.basename(m) for m in all_models]
-    model_numbers_names = [
-        (shipname.detect_model_num(m), shipname.detect_model_name(m))
-        for m in model_filenames]
+    model_numbers_names = get_models()
     latest_model = sorted(model_numbers_names, reverse=True)[0]
     return latest_model
 
 
+def enough_games_to_train():
```
Comment (Kashomon) [inline rl_loop.py]: `+` method doc

File: rl_loop.py [hunk 4]
```diff
@@ -49,20 +52,36 @@ def print_flags():
...
+    model_numbers_names = get_models()
+    if len(model_numbers_names) <= 2:
```
Comment (Kashomon) [inline rl_loop.py]: if the number of models is less then 2, then we always have enough models to train? This is a little unintuitive, so a comment would be helpful

File: rl_loop.py [hunk 5]
```diff
@@ -104,6 +123,8 @@ def gather():
 
 
 def train(logdir=None):
+    while not enough_games_to_train():
+        time.sleep(60 * 5)
```
Comment (Kashomon) [inline rl_loop.py]: How about we make the sleep-time configurable or a constant

File: rl_loop.py (hunk 4 above)
Comment (amj) [reply rl_loop.py]: there's an edge case if you only have the first model.  So either you bootstrap, and let your cluster spin for a little bit, or you just shrug and go ahead and train the first model with too little data and let it catch up for all models > 2

Comment (amj) [reply rl_loop.py]: (comment added)

File: rl_loop.py (hunk 5 above)
Comment (amj) [reply rl_loop.py]: actually, the right thing to do is exit -- otherwise the gather step won't actually be repeated.

File: rl_loop.py (hunk 1 above)
Comment (amj) [reply rl_loop.py]: Done!

File: rl_loop.py (hunk 2 above)
Comment (amj) [reply rl_loop.py]: Done!

File: rl_loop.py (hunk 3 above)
Comment (amj) [reply rl_loop.py]: Done!

**Pull request merged**